subsys/nfc/rpc/server/nfc_rpc_t2t_server.c

```diff
@@ -0,0 +1,287 @@
+	void *container = NULL;
+	size_t data_length;
+	int error = -NRF_EINVAL;
+
+	id = nrf_rpc_decode_uint(ctx);
+	data = (void *)nrf_rpc_decode_buffer_ptr_and_size(ctx, &data_length);
+
+	if (data && data_length) {
+		container = k_malloc(data_length);
+	}
+
+	if (container) {
+		memcpy(container, data, data_length);
+	}
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		nfc_rpc_report_decoding_error(NFC_RPC_CMD_T2T_PARAMETER_SET);
+		if (container) {
+			k_free(container);
+		}
+		return;
+	}
+
+	if (container) {
+		error = nfc_t2t_parameter_set(id, container, data_length);
+		k_free(container);
+	}
+
+	nrf_rpc_rsp_send_uint(group, error);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(nfc_group, nfc_rpc_t2t_parameter_set, NFC_RPC_CMD_T2T_PARAMETER_SET,
+			 nfc_rpc_t2t_parameter_set_rpc_handler, NULL);
+
+static void nfc_rpc_t2t_parameter_get_rpc_handler(const struct nrf_rpc_group *group,
+						  struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	struct nrf_rpc_cbor_ctx local_ctx;
+	nfc_t2t_param_id_t id;
+	size_t max_data_length;
+	uint8_t *container = NULL;
+	int error = -NRF_EINVAL;
+
+	id = nrf_rpc_decode_uint(ctx);
+	max_data_length = nrf_rpc_decode_uint(ctx);
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		nfc_rpc_report_decoding_error(NFC_RPC_CMD_T2T_PARAMETER_GET);
+		return;
+	}
+
+	if (max_data_length) {
+		container = k_malloc(max_data_length);
+	}
+
+	if (container) {
+		error = nfc_t2t_parameter_get(id, container, &max_data_length);
+	}
+
+	NRF_RPC_CBOR_ALLOC(&nfc_group, local_ctx, error ? 1 : max_data_length + 2);
+	nrf_rpc_encode_buffer(&local_ctx, error ? NULL : container, error ? 0 : max_data_length);
+	nrf_rpc_cbor_rsp_no_err(&nfc_group, &local_ctx);
+
+	if (container) {
+		k_free(container);
+	}
+}
+
+NRF_RPC_CBOR_CMD_DECODER(nfc_group, nfc_rpc_t2t_parameter_get, NFC_RPC_CMD_T2T_PARAMETER_GET,
+			 nfc_rpc_t2t_parameter_get_rpc_handler, NULL);
+
+static void nfc_rpc_t2t_payload_set_rpc_handler(const struct nrf_rpc_group *group,
+						struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	const void *data;
+	size_t data_length;
+	int error;
+
+	data = nrf_rpc_decode_buffer_ptr_and_size(ctx, &data_length);
+
+	if (data && data_length) {
+		memcpy(pld_buffer, data, data_length);
+	}
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		nfc_rpc_report_decoding_error(NFC_RPC_CMD_T2T_PAYLOAD_SET);
+		return;
+	}
+
+	error = nfc_t2t_payload_set(data ? pld_buffer : NULL, data_length);
+	nrf_rpc_rsp_send_uint(group, error);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(nfc_group, nfc_rpc_t2t_payload_set, NFC_RPC_CMD_T2T_PAYLOAD_SET,
+			 nfc_rpc_t2t_payload_set_rpc_handler, NULL);
+
+static void nfc_rpc_t2t_payload_raw_set_rpc_handler(const struct nrf_rpc_group *group,
+						    struct nrf_rpc_cbor_ctx *ctx,
+						    void *handler_data)
+{
+	const void *data;
+	size_t data_length;
+	int error;
+
+	data = nrf_rpc_decode_buffer_ptr_and_size(ctx, &data_length);
+
+	if (data && data_length) {
+		memcpy(pld_buffer, data, data_length);
```

Comment:
using a global buffer has thread safety implications but if we document at some point that the client can call NFC functions from a single thread only, it's OK.

Reply:
This is the requirement of NFC APIs related to payloads.
https://github.com/nrfconnect/sdk-nrfxlib/blob/main/nfc/include/nfc_t2t_lib.h#L180-L181
https://github.com/nrfconnect/sdk-nrfxlib/blob/main/nfc/include/nfc_t2t_lib.h#L214-L215